src/app.js

```diff
@@ -14,6 +14,33 @@ const formatDateForDataPoint = (dataPoint) => {
   return null;
 };
 
+const acceptableDomainReports = [
+  'site',
+  'domain',
+  'download',
+  'second-level-domain'
+];
+
+const checkDomainFilter = (req, res) => {
+  if (acceptableDomainReports.includes(req.params.reportName) &&
+  req.params.domain) {
+    return fetchData(req, res);
+  }
+  const tryReportText = acceptableDomainReports.join(', ');
+  res.status(400);
+  return res.json({
+    message: `You are requesting a report that cannot be filtered on domain. Please try one of the following reports: ${tryReportText}`,
+    status: 400
+  });
+};
+
+const filterDownloadResponse = (response, params) => {
+  if (params.domain && params.reportName === 'download') {
+    return response.filter(entry => entry.page.includes(params.domain));
```

Comment:
This may mean that a few weird urls, sneak in, like `https://example.gov/?redirect=asdf.gov`. I don't think that's a huge deal though, since parsing the urls will probs slow this down a good bit.

Reply:
hmm, yeah. I'm ok with leaving it as is for a bit until we find that it is an issue.